### PR TITLE
[Docs] Add note for date patterns used for index search.

### LINKED ISSUE #22322: Clarify use of date math in index names with custom date formats
**Elasticsearch version**: 5.1

You can run this curl command to see that date math calculation with custom formats including more than year, month and day is crippled/bogus.

```
curl -g -v 'localhost:9200/<logstash-{now%2fd{YYYY.MM.dd.hh.mm.ss}}>'

{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"logstash-2016.12.22.12.00.00","index_uuid":"_na_","index":"logstash-2016.12.22.12.00.00"}],"type":"index_not_found_exception","reason":"no such index","resource.type":"index_or_alias","resource.id":"logstash-2016.12.22.12.00.00","index_uuid":"_na_","index":"logstash-2016.12.22.12.00.00"},"status":404}
```

The index name `logstash-2016.12.22.12.00.00` does not seem to take minutes and seconds into account and the hour seems to be wrong as well.

The docs dont mention anything, that these formats are not supported - and I am not too sure they should (especially seconds), but this is unexpected behaviour so we should either fix it or document it.

### PR DESCRIPTION
**Pull request by matriv**
Add an explanatory NOTE section to draw attention to the difference
between small and capital letters used for the index date patterns.
e.g.: `HH` vs`hh`, `MM` vs `mm`.

Closes: #22322

### CODE REVIEW

File: docs/reference/api-conventions.asciidoc [hunk 1]
```diff
@@ -86,6 +86,11 @@ Where:
 `date_format`:: is the optional format in which the computed date should be rendered. Defaults to `yyyy.MM.dd`. Format should be compatible with java-time https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
 `time_zone`:: is the optional time zone. Defaults to `utc`.
 
+NOTE: Pay attention on the usage of small vs capital letters used in the `date_format`. For example:
```
Comment (romseygeek) [inline docs/reference/api-conventions.asciidoc]: s/attention on/attention to/

File: docs/reference/api-conventions.asciidoc [hunk 2]
```diff
@@ -86,6 +86,11 @@ Where:
...
+`mm` denotes minute of hour, where `MM` denotes month of year. Similarly `hh` denotes the hour in the
+`1-12` range used in combination with the `AM/PM` information, where `HH` denotes the hour in the 24-Hour
```
Comment (romseygeek) [inline docs/reference/api-conventions.asciidoc]: `1-12` range in combination with `AM/PM`, while `HH` denotes the hour in the `0-23` 24-hour range

**Review by romseygeek — APPROVED** [review]
Couple of minor language nits, but LGTM otherwise - no need for another review.

**Comment by matriv** [comment]
@romseygeek Thx! A native speaker is always better.

**Pull request merged**